### DISCUSSION TITLE
Bring back Parkour history

### DISCUSSION THREAD

**Discussion by JariRoossien** [HypixelDev/PublicAPI#438] [Ideas]
It seems the parkour history of players has been removed from the API, probably for the same as quests back then when they got removed to the last 10 to save data. I'm running into some big issues now.

- The data is flawed. Back in 2016/17 players were able to skip checkpoints and use gadgets or suits to complete parkours, this has caused a lot of the best times to be very low, inaccurate and basically useless.
- The workaround for this is using the best checkpoint times, however this brings a few issues as well. The checkpoint times aren't necessary and often not the best complete time of a player. Also this loses the completion date of the player finishing the parkour.

I understand this is yet another piece of data that could grow infinite, but there's a lot of usefulness in having the data. Even if it was the last 10 completions such as the quest storage change, it would allow me and other developers to keep track of the date when a parkour was finished, and it would allow me to not have to use cheated runs in the leaderboard.

**Comment by ConnorLinfoot** [comment]
I'm going to be slightly brutally honest here, data like this is not needed for us and very little people read it from the API. So it makes very little sense for us to keep something that can grow so large when 99% of it is useless. As the network scales we have to look at data like parkour times, quest completions, etc. As when we look at 24 million players it can really start to add up. 

We understand that some people may read this data, but at the end of the day the API is something we provide for free that displays data from our database. If we see something that is a large data set and is unnecessary it is possible for us to remove it to save storage and improve peformance.

**Reply by JariRoossien** [reply]
I completely understand. We're talking about ~1-2 billion entries probably that maybe not even 100 people use. I'm not using 99% of it either, however because of some decisions with the parkour times many years ago where the skips were allowed and those times never being resetted, I could fall back on the history to see the legitimate runs as well. I guess for now I'll just stick with the optimal times from the checkpoints to keep it sort of legitimate.

**Reply by TheNullicorn** [reply]
Out of curiosity, how often do you guys delete fields (or change their formats) across an entire collection? I get that notifying us any time anything changes is unrealistic, but I still don't understand why we can't hear about large-scale removals like this in advance, or at least without having to ask? A well-made application shouldn't be broken by these sorts of things, but it also won't know that they happened to begin with.

**Reply by ConnorLinfoot** [reply]
Technically the data structure can change every day, and can be done for all sorts of reasons. It's really not plausible to have each developer announce somewhere when something is changed that results in player data.
